lib/api/v1/source-controller.js

```diff
@@ -60,9 +74,12 @@ module.exports = function(app, user) {
   // Reset unread error count
   app.put('/api/v1/source/_events/:_id', user.can('edit sources'), function(req, res) {
     Source.resetUnreadErrorCount(req.params._id, function(err, source, numberAffected) {
-      if (err) res.send(err.status, err.message);
-      else if (!source) res.send(404);
-      else res.send(200, source);
+      if (err) return res.send(err.status, err.message);
+      else if (!source) return res.send(404);
+      Source.populate(source, [{path: 'user', select: 'username'}], function(err, source){
+        if (err) res.send(err.status, err.message);
+        else res.send(200, source);
```

Comment:
Not sure if it's necessary to populate here. Or even to return the source. Can you check if the client side code is actually using the returned source here?